taxcalc/records.py

```diff
@@ -240,45 +237,18 @@ def __init__(self,
                  weights=weights_path,
                  start_year=None,
                  **kwargs):
-
-        self.read(data)
-        self.read_blowup(blowup_factors)
-        self.read_weights(weights)
-        if (start_year):
+        """
+        Records class constructor
+        """
+        self._read_data(data)
+        self._read_blowup(blowup_factors)
+        self._read_weights(weights)
+        if start_year:
             self._current_year = start_year
         else:
             self._current_year = self.FLPDYR[0]
-
-        """Imputations"""
-        self._cmbtp_itemizer = None
-        self._cmbtp_standard = self.e62100 - self.e00100 + self.e00700
-        self.mutate_imputations()  # Updates the self._cmbtp_itemizer variable
-
-        # Standard deduction amount in 2009
-        std2009 = np.array([5700, 11400, 5700, 8350, 11400, 5700, 950])
-        # Additional standard deduction for aged 2009
-        STD_Aged_2009 = np.array([1400., 1100.])
-        # Compulsory itemizers
-        self._compitem = np.where(np.logical_and(self.FDED == 1,
-                                                 self.e04470 <
-                                                 std2009[self.MARS - 1]), 1, 0)
-        # Number of taxpayers
-        self._txpyers = np.where(np.logical_or(self.MARS == 2,
-                                               np.logical_or(self.MARS == 3,
-                                                             self.MARS == 6)),
-                                 2., 1.)
-        # Number of extra standard deductions for aged
-        self._numextra = np.where(np.logical_and(self.FDED == 2, self.e04470 <
-                                  std2009[self.MARS - 1]),
-                                  np.where(
-                                  np.logical_and(self.MARS != 2,
-                                                 self.MARS != 3),
-                                  (self.e04470 - std2009[self.MARS - 1]) /
-                                  STD_Aged_2009[0],
-                                  (self.e04470 - std2009[self.MARS - 1]) /
-                                  STD_Aged_2009[1]),
-                                  np.where(self.e02400 > 0, self._txpyers, 0))
-        self._puf_year = 2009
+        if self._current_year == 2009:
+            self._impute_variables()
```

Comment:
Similarly here, if the _current_year was set to some other year other 2009 due to the start_year attribute, we're not doing the imputations anymore. Should we add a warning message in an else loop?